pkg/api/validation/validation.go

```diff
@@ -51,3 +53,37 @@ func validateSecret(secret *corev1.Secret, fldPath *field.Path) field.ErrorList
 
 	return allErrs
 }
+
+func validateOnmetalMachineClassSpec(spec *v1alpha1.ProviderSpec, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	emptyRootDisk := resource.Quantity{}
+	if spec.RootDisk.Size == emptyRootDisk {
```

Comment:
Root disk is an optional field as it can be empty - see other comment.

Reply:
Addressed.